### ISSUE TITLE
Creating front door origin fails with message "Endpoint pools must contain at least one enabled endpoint"

### ISSUE THREAD

**Issue by Masahigo** [hashicorp/terraform-provider-azurerm#18333]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.21.1

### Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_origin

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "rg-example"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                      = "examplestoracc"
  resource_group_name       = azurerm_resource_group.example.name
  location                  = azurerm_resource_group.example.location
  account_kind              = "StorageV2"
  account_tier              = "Standard"
  account_replication_type  = "LRS"
  enable_https_traffic_only = true
  min_tls_version           = "TLS1_2"

  network_rules {
    default_action = "Deny"
    ip_rules       = local.ip_whitelist
  }
}

resource "azurerm_cdn_frontdoor_profile" "example" {
  name                     = "frontdoor-profile-example"
  resource_group_name      = azurerm_resource_group.example.name
  response_timeout_seconds = 60
  sku_name                 = "Standard_AzureFrontDoor"
}

resource "azurerm_cdn_frontdoor_endpoint" "example" {
  name                     = "frontdoor-endpoint-example"
  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id
}

resource "azurerm_cdn_frontdoor_origin_group" "example" {
  name                     = "default-origin-group"
  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id

  load_balancing {}
}

resource "azurerm_cdn_frontdoor_origin" "example" {
  name                          = "default-origin"
  cdn_frontdoor_origin_group_id = azurerm_cdn_frontdoor_origin_group.example.id

  health_probes_enabled          = false
  certificate_name_check_enabled = true

  host_name          = azurerm_storage_account.example.primary_blob_host
  origin_host_header = azurerm_storage_account.example.primary_blob_host
  priority           = 1
  weight             = 1000
}
```

### Debug Output/Panic Output

```shell
Error: waiting for the creation of Front Door Origin: (Origin Name "default-origin" / Origin Group Name "default-origin-group" / Profile Name "frontdoor-profile-example" / Resource Group "rg-example"): Code="BadRequest" Message="{\"ErrorMessage\":\"Endpoint pools must contain at least one enabled endpoint\"}"
```

### Expected Behaviour

The origin should have gotten created

### Steps to Reproduce

1. terraform init
2. terraform plan -out tfplan
3. terraform apply tfplan

**Comment by Masahigo** [comment]
I was following the example from TF documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_frontdoor_origin

Expected end result would be a new origin group with "Unassociated" routes: https://docs.microsoft.com/en-us/azure/frontdoor/how-to-configure-origin

**Comment by JakeGinnivan** [comment]
I've spent the last few hours trying to figure out what is going on here. 

I have hit the same error in the portal, but doesn't seem to happen as often. It also happens using azapi, and I'd assume bicep etc too.

If you hit the error, then delete the origin group which exists in the portal (it is in a bad state, and can't be used), then try again, eventually it will be successfully created.

It looks like it's a bug in frontdoor itself?

**Comment by WodansSon** [comment]
@JakeGinnivan thank you for opening this issue, I believe you are hitting issue #18063. In your origin resource try changing the `health_probes_enabled` to `true`. There was an API documentation issue that accidentally identified the origin's `enable`/`disable` field as controlling if the state of the health probes. I have fixed the attributes name in PR #18231 and deprecated the existing `health_probes_enabled` field for the `enabled` field.

**Comment by WodansSon** [comment]
@Masahigo I am unable to reproduce this issue locally using the provided example HCL configuration and the v3.21.1 provider. At this point I would tend to agree with @JakeGinnivan, that this is in fact a bug in the service itself and not an issue with the Terraform provider. The error that you mention in this issue is not coming from Terraform that is coming from the Service. I will continue to investigate this issue to see if I can get a repro.

**Comment by Masahigo** [comment]
Thanks for the tips @JakeGinnivan and @WodansSon!

I managed to create it successfully now, by setting `health_probes_enabled` to `true` for the **azurerm_cdn_frontdoor_origin** resource and configuring the `health_probe` like so

```hcl

resource "azurerm_cdn_frontdoor_origin_group" "frontdoor" {
...
 health_probe {
    interval_in_seconds = 100
    path                = "/"
    protocol            = "Https"
    request_type        = "HEAD"
  }
}
```

But it wasn't enough to delete the origin group via Azure portal. I needed to scratch the whole Azure Front Door Standard resource and re-provision from scratch with these additions and then it worked.
